drafts/bdd-specflow.md

```diff
@@ -0,0 +1,187 @@
+
+To generate the binding file add the feature file to your project (or write it from scratch), open it, right-click in the editor and choose **Define Steps**. This command will create the file for you.
+
+![Generating step definitions for a feature file](https://cdn.hashnode.com/res/hashnode/image/upload/v1682673140883/1FwNlUtLx.webp?auto=format)
+
+***Generating step definitions for a feature file***
+
+It is good practice to have one binding file per feature file. If you add tests later, then the **Copy to clipboard** button is your friend which will allow you to insert the bindings into an existing file.
+
+## Writing the tests
+
+The binding file contains methods without a body (or just a placeholder). The task is now to fill the body with a reasonable test code.
+A `Given` method is supposed to prepare data like setting up a database or starting a service. A `When` method should perform user or system actions like clicking a button or writing text in fields. A `Then` method finally ensures that the outcome of the action combined with the preconditions fulfills the expected behavior. See the code example below for inspiration.
+
+```csharp
+[Binding]
+public class LoginStepDefinitions
+{
+    private readonly MyApp _app = new MyApp();
+    private Exception? _exception;
+
+    [Given(@"the user has opened the login page")]
+    public void GivenTheUserHasOpenedTheLoginPage()
+    {
+        _app.NavigationService.GoToLoginPage();
+    }
+
+    [When(@"the user enters invalid login credentials")]
+    public void WhenTheUserEntersInvalidLoginCredentials()
+    {
+        _app.LoginService.EnterCredentials("nope", "nope");
+    }
+
+    [When(@"the user clicks on the login button")]
+    public void WhenTheUserClicksOnTheLoginButton()
+    {
+        try
+        {
+            _app.LoginService.Login();
+            _app.NavigationService.GoToInternalPage();
+        }
+        catch (Exception ex)
+        {
+            _exception = ex;
+        }
+    }
+
+    [Then(@"an error message is shown")]
+    public void ThenAnErrorMessageIsShown()
+    {
+        Assert.IsNotNull(_exception);
+    }
+
+    [Then(@"the user stays on the login page")]
+    public void ThenTheUserStaysOnTheLoginPage()
+    {
+        Assert.IsTrue(_app.NavigationService.IsOnLoginPage);
+    }
+
+    [When(@"the user enters valid login credentials")]
+    public void WhenTheUserEntersValidLoginCredentials()
+    {
+        _app.LoginService.EnterCredentials("admin", "pass");
+    }
+
+    [Then(@"the user is redirected to an internal page")]
+    public void ThenTheUserIsRedirectedToAnInternalPage()
+    {
+        Assert.IsTrue(_app.NavigationService.IsOnInternalPage);
+    }
+}
+```
+
+## Executing the tests
+
+Build your project and then the test runner should find the new tests and be able to execute them. For every feature file, there will an auto-generated code after the build.
```

Comment:
It seems like there is a word missing after the comma: `there will an auto-generated code`. 
```suggestion
Build your project and then the test runner should find the new tests and be able to execute them. For every feature file, there will be an auto-generated code after the build.
```

Reply:
I even added two words :)

Reply:
Perfect 😅👍